internal/pgbackrest/config_test.go

```diff
@@ -23,284 +20,161 @@ import (
-					Repos: []v1beta1.PGBackRestRepo{{
-						Name:   "repo1",
-						Volume: &v1beta1.RepoPVC{},
-					}, {
-						Name: "repo2",
-						Azure: &v1beta1.RepoAzure{
-							Container: "container",
-						},
-					}, {
-						Name: "repo3",
-						GCS: &v1beta1.RepoGCS{
-							Bucket: "bucket",
-						},
-					}, {
-						Name: "repo4",
-						S3: &v1beta1.RepoS3{
-							Bucket:   "bucket",
-							Endpoint: "endpoint",
-							Region:   "region",
-						},
-					}},
-				},
+		{
+			Name: "repo3",
+			GCS:  &v1beta1.RepoGCS{Bucket: "g-bucket"},
+		},
+		{
+			Name: "repo4",
+			S3: &v1beta1.RepoS3{
+				Bucket: "s-bucket", Endpoint: "endpoint-s", Region: "earth",
 			},
 		},
 	}
 
-	// the initially created configmap
-	var cmInitial *corev1.ConfigMap
-	// the returned configmap
-	var cmReturned corev1.ConfigMap
-	// pod spec for testing projected volumes and volume mounts
-	pod := &corev1.PodSpec{}
-
-	testInstanceName := "test-instance-abc"
-	testRepoName := "repo-host"
-	testConfigHash := "abcde12345"
-
-	domain := naming.KubernetesClusterDomain(context.Background())
-
-	t.Run("pgbackrest configmap checks", func(t *testing.T) {
-
-		// setup the test environment and ensure a clean teardown
-		testEnv, testClient := setupTestEnv(t)
-
-		// define the cleanup steps to run once the tests complete
-		t.Cleanup(func() {
-			teardownTestEnv(t, testEnv)
-		})
-
-		t.Run("create pgbackrest configmap struct", func(t *testing.T) {
-			// create an array of one host string value
-			pghosts := []string{testInstanceName}
-			// create the configmap struct
-			cmInitial = CreatePGBackRestConfigMapIntent(postgresCluster, testRepoName,
-				testConfigHash, naming.ClusterPodService(postgresCluster).Name, "test-ns", pghosts)
-
-			// check that there is configmap data
-			assert.Assert(t, cmInitial.Data != nil)
-		})
-
-		t.Run("create pgbackrest configmap", func(t *testing.T) {
-
-			ns := &corev1.Namespace{}
-			ns.Name = naming.PGBackRestConfig(postgresCluster).Namespace
-			ns.Labels = labels.Set{"postgres-operator-test": ""}
-			assert.NilError(t, testClient.Create(context.Background(), ns))
-			t.Cleanup(func() { assert.Check(t, testClient.Delete(context.Background(), ns)) })
-
-			// create the configmap
-			err := testClient.Patch(context.Background(), cmInitial, client.Apply, client.ForceOwnership, client.FieldOwner(testFieldOwner))
-
-			assert.NilError(t, err)
-		})
-
-		t.Run("get pgbackrest configmap", func(t *testing.T) {
-
-			objectKey := client.ObjectKey{
-				Namespace: naming.PGBackRestConfig(postgresCluster).Namespace,
-				Name:      naming.PGBackRestConfig(postgresCluster).Name,
-			}
-
-			err := testClient.Get(context.Background(), objectKey, &cmReturned)
-
-			assert.NilError(t, err)
-		})
-
-		// finally, verify initial and returned match
-		assert.Assert(t, reflect.DeepEqual(cmInitial.Data, cmReturned.Data))
+	configmap := CreatePGBackRestConfigMapIntent(cluster,
```

Comment:
:pick: I'm 50/50 here, but I feel like this would be a good place to start a `t.Run` section: everything that comes before is setup that is used by the two test cases; everything that comes after are the two cases: "standard in every way", and "custom metadata."

Reply:
Amended in [`2b0a8e015`](https://github.com/CrunchyData/postgres-operator/compare/2c4048ec8410e68e476b13997acf673ae5ab6b8b..2b0a8e015a4a54bfc5c745f533531228d3bdae10). I named the subtest `DedicatedRepoHost` because that is what is really being tested here. The two subtests now make a copy of `cluster` because they are independent.